src/Writers/LaTeXWriter.jl

```diff
@@ -214,54 +215,25 @@ end
 function writeheader(io::IO, doc::Documents.Document)
     custom = joinpath(doc.user.root, doc.user.source, "assets", "custom.sty")
     isfile(custom) ? cp(custom, "custom.sty"; force = true) : touch("custom.sty")
+    
+    custom_preamble_file = joinpath(doc.user.root, doc.user.source, "assets", "preamble.tex")
+    if isfile(custom_preamble_file)
+        # copy custom preamble.
+        cp(custom_preamble_file, "preamble.tex"; force = true)
+    else # no custom preamble.tex, use default.
+        cp(DEFAULT_PREAMBLE_PATH, "preamble.tex"; force = true)
+    end
     preamble =
         """
-        \\documentclass[oneside]{memoir}
-
-        \\usepackage{./documenter}
-        \\usepackage{./custom}
-
-        %% Title Page
-        \\title{
-            {\\HUGE $(doc.user.sitename)}\\\\
-            {\\Large $(get(ENV, "TRAVIS_TAG", ""))}
-        }
-        \\author{$(doc.user.authors)}
-
-        %% TOC settings
-        % -- TOC depth
-        %   value: [part, chapter, section, subsection,
-        %           subsubsection, paragraph, subparagraph]
-        \\settocdepth{section}  % show "part+chapter+section" in TOC
-        % -- TOC spacing
-        %   ref: https://tex.stackexchange.com/questions/60317/toc-spacing-in-memoir
-        %   doc: memoir/memman.pdf
-        %       - Figure 9.2: Layout of a ToC
-        %       - Table 9.3: Value of K in macros for styling entries
-        \\makeatletter
-        % {part} to {chaper}
-        \\setlength{\\cftbeforepartskip}{1.5em \\@plus \\p@}
-        % {chaper} to {chaper}
-        \\setlength{\\cftbeforechapterskip}{0.0em \\@plus \\p@}
-        % Chapter num to chapter title spacing (Figure 9.2@memman)
-        \\setlength{\\cftchapternumwidth}{2.5em \\@plus \\p@}
-        % indent before section number
-        \\setlength{\\cftsectionindent}{2.5em \\@plus \\p@}
-        % Section num to section title spacing (Figure 9.2@memman)
-        \\setlength{\\cftsectionnumwidth}{4.0em \\@plus \\p@}
-        \\makeatother
-
-        %% Main document begin
-        \\begin{document}
-
-        \\frontmatter
-        \\maketitle
-        \\clearpage
-        \\tableofcontents
-
-        \\mainmatter
-
+        % Useful variables
+        \\newcommand{\\DocMainTitle}{$(doc.user.sitename)}
+        \\newcommand{\\DocVersion}{$(get(ENV, "TRAVIS_TAG", ""))}
```

Comment:
Since we rarely use `TRAVIS` anymore, should we just set the `\DocVersion` variable to `VERSION`?
```suggestion
        \\newcommand{\\DocVersion}{ $(VERSION) }
```

Reply:
Travis is definitely gone. @mortenpi ok?

Reply:
I just realized that `VERSION` refers to the version of julia, but for users of Documenter.jl, it's more like they want the version of their own package.

Maybe this variable should be deprecated or deleted.
If the user needs the version number to be displayed, just append it to the site name.

Reply:
I was just thinking the same thing.

Reply:
> `TRAVIS_TAG`: If the current build is for a git tag, this variable is set to the tag’s name, otherwise it is empty (`""`).
> [Environment Variables - Travis CI](https://docs.travis-ci.com/user/environment-variables/)

And
https://github.com/JuliaDocs/Documenter.jl/blob/cd49b078a8ce6e1ff17f56d8b98f3f92e93b963f/src/deployconfig.jl#L136-L139

Reply:
So this would just try to determine the package version number automatically, but only works if it's being used on Travis (or if the user sets the environment variable themselves; e.g. it used to work [for Documenter](https://github.com/JuliaDocs/Documenter.jl/blob/gh-pages-pdf/v0.21.0/Documenter.jl-0.21.0.pdf)). However, I don't think it's actually good that it's quietly relying on the enviroment variable.

I would suggest adding another `version` variable to the `LaTeX` object. To make this non-breaking and non-behavior-changing, it can still just default to `get(ENV, "TRAVIS_TAG", "")`, but I think it should be documented that this default is deprecated. On the Julia side, we can then populate this new option so that the resulting PDFs would actually contain a version number.